src/extreme_filter.jl

```diff
@@ -268,3 +267,206 @@ Base.@propagate_inbounds function _minimum_fast(A::AbstractArray{Bool}, p, Ω)
     end
     return rst
 end
+
+function _padded_copyto!(dest::AbstractVector, src::AbstractVector, shift, v)
+    Base.require_one_based_indexing(dest, src)
+    if shift >= 0
+        o = shift
+        vsrc = @view src[(1+o):end]
+        N = length(vsrc)
+        copyto!(dest, 1, vsrc, 1, N)
+        vdest = @view dest[(N+1):end]
+        fill!(vdest, v)
+    else
+        o = -shift
+        vdest = @view dest[1:o]
+        fill!(vdest, v)
+        copyto!(dest, 1 + o, src, 1, length(dest) - o)
+    end
+    return dest
+end
+
+# ptr level optimized implementation for Real types
+# short-circuit all check
+function _unsafe_padded_copyto!(dest::AbstractVector, src::AbstractVector, dir, N, v) where {T}
+    if dir
+        unsafe_store!(pointer(dest), v)
+        unsafe_copyto!(pointer(dest, 2), pointer(src, 1), N - 1)
+    else
+        unsafe_copyto!(pointer(dest, 1), pointer(src, 2), N - 1)
+        unsafe_store!(pointer(dest), v, N)
+    end
+    return dest
+end
+
+# ptr level optimized implementation for Real types
+# short-circuit all check
+# call LoopVectorization directly
+function _unsafe_shift_arith!(f, out::AbstractArray{T,1}, tmp::AbstractArray{T,1}, A::AbstractArray{T,1}) where {T} #tmp external to reuse external allocation
+    if f === min
+        padd = typemax(T)
+    else
+        padd = typemin(T)
+    end
+    N = length(out)
+    _unsafe_padded_copyto!(tmp, A, true, N, padd)
+    LoopVectorization.vmap!(f, out, A, tmp)
+    _unsafe_padded_copyto!(tmp, A, false, N, padd)
+    LoopVectorization.vmap!(f, out, out, tmp)
```

Comment:
If we introduce an additional buffer array, we can also fuse the two vmap into one loop

```julia
    _unsafe_padded_copyto!(tmp, A, true, N, padd)
    _unsafe_padded_copyto!(tmp2, A, false, N, padd)

    @turbo for i in eachindex(out, A, tmp, tmp2)
        out[i] = f(f(A[i], tmp[i]), tmp2[i])
    end

Reply:
I'm not aware of this "trick" for sure i will maybee avoid copy and enforce cache locallity